src/robotFactory.js

```diff
@@ -1,10 +1,85 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.chipVersion = chipVersion;
+    this.coords = coords;
 
-class FlyingRobot {}
+    if (!coords.x) {
+      this.coords.x = 0;
+    } else {
+      this.coords.x = coords.x;
+    }
 
-class DeliveryDrone {}
+    if (!coords.y) {
+      this.coords.y = 0;
+    } else {
+      this.coords.y = coords.y;
+    }
+  }
+
+  goForward(step = 1) {
+    this.coords.y += step;
+  }
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+  }
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  }
+
+  goRight(step = 1) {
+    this.coords.x += step;
+  }
+
+  getInfo() {
+    return `Robot: ${
+      this.name}, Chip version: ${this.chipVersion}, Weight: ${this.weight}`;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    super(name, weight, coords, chipVersion);
+
+    if (!coords.z) {
+      this.coords.z = 0;
+    } else {
+      this.coords.z = coords.z;
+    }
+  }
+
+  goUp(step = 1) {
+    this.coords.z += step;
+  }
+
+  goDown(step = 1) {
+    this.coords.z -= step;
+  }
+}
+
+class DeliveryDrone extends FlyingRobot {
+  constructor(name, weight, coords, chipVersion,
+    maxLoadWeight, currentLoad = null) {
```

Comment:
No need to add a default value to currentLoad
```suggestion
  constructor(
    name, 
    weight, 
    coords, 
    chipVersion,
    maxLoadWeight, 
    currentLoad
  ) {
```

Reply:
Without default value the test is failed - screenshot.
<img width="940" alt="Снимок экрана 2023-02-27 в 15 48 28" src="https://user-images.githubusercontent.com/74835422/221568751-2bfd016d-8144-4579-90b3-d7dfc92b54c9.png">